### PR TITLE
Refactor the frontend code

### PR DESCRIPTION
**Pull request by HenryChen233**
- Refactor the code including using only one client
- Reuse the code to decode and encode JSON
- Rename some of the Struct and functions to make it more clear
- Add more complete function on the home page

### CODE REVIEW

File: request_handler.go [hunk 1]
```diff
@@ -388,16 +417,138 @@ func loginHandler(w http.ResponseWriter, r *http.Request) {
...
 
+func editEmailHandler(w http.ResponseWriter, r *http.Request){
+	cookie, err := r.Cookie("V")
+	if err != nil {
+		log.Println("cookie may expire login again", err)
+		// When refactoring the code, a more specific page may needed
+		http.Redirect(w, r, "/loginError/", http.StatusFound)
+		return
+	}
+	cookieValue := strings.TrimPrefix(cookie.Value, "cookie=")
```
Comment (sleepingpig) [inline request_handler.go]: Why the cookie value has prefix "cookie="? It doesn't look correct.

File: request_handler.go [hunk 2]
```diff
@@ -388,16 +417,138 @@ func loginHandler(w http.ResponseWriter, r *http.Request) {
...
+	newEmail := r.FormValue("email")
+	token := r.FormValue("CSRFToken")
+	// Get cookie from browser
+	cookie, err := r.Cookie("V")
+	if err != nil {
+		log.Println("login expire, log in again", err)
+		http.Redirect(w, r, "/home/", http.StatusFound)
+		return
+	}
+	cookieValue := strings.TrimPrefix(cookie.Value, "cookie=")
```
Comment (sleepingpig) [inline request_handler.go]: Why the cookie value has prefix "cookie="? It doesn't look correct.

File: request_handler.go (hunk 1 above)
Comment (HenryChen233) [reply request_handler.go]: It was sent by backend, and this is to trim it.

File: request_handler.go [hunk 3]
```diff
@@ -200,45 +222,50 @@ func saveEditedInfo(w http.ResponseWriter, r *http.Request, title string) {
...
 		http.Redirect(w, r, "/loginError/", http.StatusFound)
 		return
 	}
-	request.Header.Add("Cookie", cookieValue)
 	request.Header.Add("Content-Type", "application/json")
-	_, err = client.Do(request)
-	client.CloseIdleConnections()
+	request.Header.Add("Cookie", cookieValue)
+	resp, err := client.Do(request)
+	log.Println(resp)
```
[the remark this answers is not in the document]
Comment (HenryChen233) [reply request_handler.go]: This is by mistake when merging, would correct it.

File: request_handler.go (hunk 2 above)
Comment (DandilionLau) [reply request_handler.go]: It is partially my fault at the backend, that I did not change the cookie keyName before the previous PR. Now we should use `V` for the key everywhere.

File: request_handler.go (hunk 1 above)
Comment (DandilionLau) [reply request_handler.go]: It is partially my fault at the backend, that I did not change the cookie keyName before the previous PR. Now we should use V for the key everywhere.

**Comment by DandilionLau** [comment]
Maybe we should add a line of running unit tests to `.travis.yml`.

**Comment by DandilionLau** [comment]
Maybe changing `public_profile.html` to  `publicProfile.html` to make it consistent with other templates?

**Comment by sleepingpig** [comment]
I've combine your changes in this PR + some fix for cookies handling logic into https://github.com/verizon-media-2019-ais3/identity-frontend/pull/9

**Pull request closed without merging**